### DISCUSSION TITLE
[BE101]  echo出的標籤特性，外部CSS無法套用

### DISCUSSION THREAD

**Discussion by IamTomIamQQ** [Lidemy/forum#126] [後端相關問題]
參考board/index.php等
撰寫留言板時，每次功能頁若無法寫入成功時會回傳給前端訊息告知錯誤與顯示，如下

PHP
```
<?php
        if (!empty($_GET['errCode'])) {
          $code = $_GET['errCode'];
          $msg = 'Error';
          if ($code === '1') {
            $msg = '資料不齊全';
          }
          echo '<h2 class="error">錯誤：' . $msg . '</h2>';
        }
      ?>
```

CSS
```
.error {
　color: red;
}
```
但如上CSS寫法，.error選擇器都無法套用，最後我發現我用直接法撰寫才能套用此屬性(如下)
```
**echo '<h2 class="error" style ="color : red">錯誤：' . $msg . '</h2>'**
```
## 但，這是為什麼呢?

到底漏了什麼呢?
還是說這跟SERVER設定有關呢?

**Comment by allenliaooooooo** [comment]
有試過清掉 Cache 嗎？

**Comment by outshaker** [comment]
不確定你附的 php 程式碼是部分還是全部，按照現有條件推測可能的原因：
1. 你的 php 沒有指定引用外部 CSS 樣式表
2. 你引用的 CSS 樣式表路徑可能有錯

參考寫法：

```php
<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <title></title>
  <link rel="stylesheet" href="這邊放你的CSS樣式表路徑">
</head>
<body>
  <?php
    if (!empty($_GET['errCode'])) {
      $code = $_GET['errCode'];
      $msg = 'Error';
      if ($code === '1') {
        $msg = '資料不齊全';
      }
      echo '<h2 class="error">錯誤：' . $msg . '</h2>';
    }
  ?>
</body>
</html>
```

**Reply by IamTomIamQQ** [reply]
**SORRY** 
 一切都是我的錯 ! 我自己親自把自己送入鬼故事中了。

照著原來自己的思路不知道為什麼我就一直卡在鬼打牆的地方。
有關於檢查過程

> 1. Cache Disable有打勾了。
> 2. 外部樣式引入應該是沒錯，因為其他基本樣式沒跑掉。

### 但我就是鬼遮眼了

　　我一直沒發現class拼錯字。改好class後起初還是沒套用到，`當我試著來回勾 Cache Disable，然後點到clear request時`顏色就能套上了。說真的我還是一頭霧水。

PS 我怎麼發現一定是我自己寫錯呢? 其實我把老師的index與style複製過來做一份新的發現根本沒問題....(Q..Q) 覺得有點丟臉
